src/client/mod.rs

```diff
@@ -48,11 +47,23 @@ impl<P: Provider> Client<P> {
     /// );
     /// ```
     pub fn new(client_id: String, client_secret: String, redirect_uri: Option<String>) -> Self {
+        Client::with_provider(client_id, client_secret,  P::default(), redirect_uri)
+    }
+}
+
+impl<P: Provider> Client<P> {
+    /// Creates a client with a given Provider. Use when the provider needs non-default Initialization.
+    pub fn with_provider(
+            client_id: String, 
+            client_secret: String, 
+            provider: P, 
+            redirect_uri: Option<String>
```

Comment:
Might be worth re-evaluating the initialization API now that there are essentially two optional parameters.